### DISCUSSION TITLE
Document rendering on localhost using RStudio

### DISCUSSION THREAD

**Discussion by danielrmt** [quarto-dev/quarto-cli#2815] [Q&A]
When I render a qmd document using the "Render" button in RStudio, the document is rendered fine, but RStudio starts a temporary webserver and opens the document in localhost. I just want to open the document in my normal browser (for HTML) or PDF viewer (for PDF) with no server at all.  How to change this behavior?

**Comment by rgaiacs** [comment]
Thanks for the question @danielrmt. 

> I just want to open the document in my normal browser (for HTML) or PDF viewer (for PDF) with no server at all. How to change this behavior?

For the HTML version, you can select between `Window` and `Viewer pane`.

![vlcsnap-2022-10-13-09h41m19s517](https://user-images.githubusercontent.com/1506457/195479300-5962f310-6560-4fdb-959d-0b6a21d00b5e.png)

For the PDF version, you can select between `RStudio Viewer` and `System viewer`.

![vlcsnap-2022-10-13-09h41m54s886](https://user-images.githubusercontent.com/1506457/195479307-ed98ab28-0a09-4635-8ccf-9ede9f849768.png)

**Comment by danielrmt** [comment]
Thank you for your response. I already had HTML output at "Window", and PDF output was at "(No Preview)". Changed it  to "System viewer". This seems to works fine for Rmd documents, but for qmd documents, RStudio still serves a local HTTP server after compiling the document.

**Reply by rgaiacs** [reply]
> This seems to works fine for Rmd documents, but for qmd documents, RStudio still serves a local HTTP server after compiling the document.

I don't think you will be able to change this behaviour. The `render` button is configured to run `quarto preview` which uses a local HTTP server. Instead of click the `render` button, you can run `quarto build` in bash or `quarto_render()` in R (see https://quarto.org/docs/tools/rstudio.html#r-package for details).

**Comment by danielrmt** [comment]
Ok, thanks!
